### ISSUE TITLE
Add an option to train with chosen reward score model and potential injected noise in the training dataset

### ISSUE THREAD

**Issue by mauriciogtec** [vermashresth/RLAIF-KL-Error#12]
### Change plan

- The cdpo cli needs to be trained to take in an optional reward model to resample labels
- When the reward model is given, it is assumed that the dataset has a column bt_prob_{model}, the latter can be used to sample labels, which are needed for SFT and DPO, etc.
- Make sure the sft and dpo use the sampled labels, and other pipelines
- For the noise types, include two types (each with potential additional parameters). One type is label switching with random noise. The other is adding noise to bt_prob, something like ( sigmoid(logit(bt_prob) + noise))
- The types of noise should be configured in a new config, so the config files need to be updated as well. The default noise should be zero, and nothing happens.
- The saved models along the pipeline should be tagged so that the type of input reward model and noise level if given, are part of the scripts, run names, logs, etc.
